lib/src/models/data_bundle.dart

```diff
@@ -41,43 +42,84 @@ class DataBundle {
 
   static Future<DataBundle> fromHandles({
     required List<DataItemHandle> handles,
+    bool parallelize = true,
   }) async {
+    log('Get DataBundle from handles...');
+    log('Number of handles: ${handles.length}');
+
     final noOfDataItems = handles
         .map((handle) => handle.dataItemCount)
         .reduce((sum, current) => sum += current);
     final headers = Uint8List(64 * noOfDataItems);
     // Use precalculated buffers if provided to
     var dataItemIndex = 0;
     final binaries = BytesBuilder();
-    await Future.wait(
-      handles.map(
-        (handle) async {
-          // Sign DataItem
-          final dataItems = await handle.getDataItems();
-          for (var dataItem in dataItems) {
-            final id = decodeBase64ToBytes(dataItem.id);
-            // Create header array
-            final header = Uint8List(64);
-            final raw = await dataItem.asBinary();
-            dataItem.data = Uint8List(0);
-            // Set offset
-            header.setAll(0, longTo32ByteArray(raw.length));
-            // Set id
-            header.setAll(32, id);
-            // Add header to array of headers
-            headers.setAll(64 * dataItemIndex, header);
-            dataItemIndex++;
-            // Convert to array for flattening
-            binaries.add(raw.takeBytes());
-          }
-        },
-      ),
-    );
+
+    Future<void> getDataItems(DataItemHandle handle) async {
+      log('current ${handle.hashCode}');
+      // Sign DataItem
+      log('Getting DataItems');
+
+      final dataItems = await handle.getDataItems();
+
+      log('DataItems done');
+      log('Number of data items: ${dataItems.length}');
+
+      log('Iterating for each DataItem');
+
+      for (var dataItem in dataItems) {
+        log('Current dataItem: ${dataItem.id}');
+
+        log('Decoding');
+
+        final id = decodeBase64ToBytes(dataItem.id);
+
+        log('Decoded');
+
+        // Create header array
+        final header = Uint8List(64);
+
+        log('Get raw');
+
+        final raw = await dataItem.asBinary();
+
+        log('Raw done: ${raw.length}');
+
+        dataItem.data = Uint8List(0);
+        // Set offset
+        header.setAll(0, longTo32ByteArray(raw.length));
+        // Set id
+        header.setAll(32, id);
+        // Add header to array of headers
+        headers.setAll(64 * dataItemIndex, header);
+        dataItemIndex++;
+        // Convert to array for flattening
+        binaries.add(raw.takeBytes());
+
+        log('Added binaries from ${dataItem.id}');
+      }
+    }
+
+    if (parallelize) {
```

Comment:
We could have a maxConcurrentFutures int argument. Not a blocker, just a comment